web/src/pages/api/admin/otel-ingestion-replay.ts

```
@@ -0,0 +1,194 @@
+import { type NextApiRequest, type NextApiResponse } from "next";
+import { z } from "zod/v4";
+import { randomUUID } from "crypto";
+import {
+  logger,
+  traceException,
+  instrumentAsync,
+  OtelIngestionQueue,
+  QueueJobs,
+  getS3EventStorageConfig,
+  listS3FilesPaginated,
+  generateOtelS3Prefixes,
+} from "@langfuse/shared/src/server";
+import { AdminApiAuthService } from "@/src/ee/features/admin-api/server/adminApiAuth";
+
+/*
+This API route is used to replay OTEL ingestion events from S3 for a specific project
+within a given time range. It lists S3 objects and queues them for reprocessing.
+*/
+
+const RequestSchema = z.object({
+  projectId: z.string(),
+  startDate: z.string().datetime(),
+  endDate: z.string().datetime(),
+  granularity: z.enum(["hour", "minute"]).default("hour"),
+});
+
+const MAX_FILES = 1_000_000;
+const BATCH_SIZE = 10_000;
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  try {
+    // Only allow POST
+    if (req.method !== "POST") {
+      return res.status(405).json({ error: "Method Not Allowed" });
+    }
+
+    // Auth check - allowed on Langfuse Cloud
+    if (
+      !AdminApiAuthService.handleAdminAuth(req, res, {
+        isAllowedOnLangfuseCloud: true,
+      })
+    ) {
+      return;
```

We should return an error here